activerecord/lib/active_record/associations/has_one_association.rb

```diff
@@ -81,15 +81,17 @@ def remove_target!(method)
             target.delete
           when :destroy
             target.destroyed_by_association = reflection
-            target.destroy
+            touch_owner_if(target.persisted?) do
```

Comment:
Is there any reason to call `destroy` if the target is not persisted?

Reply:
@rafaelfranca no, there's not; just an oversight. resolving :)

Reply:
@rafaelfranca probably not! Will resolve :)